jax/lax/lax.py

```diff
@@ -1320,6 +1320,17 @@ def slice_in_dim(operand, start_index, limit_index, stride=1, axis=0):
   limit_indices = list(operand.shape)
   strides = [1] * operand.ndim
 
+  # translate `None`
```

Comment:
I'm wary of changing the behavior of `lax.slice_in_dim` here just for implementing a single numpy function. An alternative is just to write a helper function in the body of `_gradient` with this behavior.

Can you make the change in `_gradient` instead, basically adding a helper function instead of using `partial(lax.slice_in_dim, ...)` there?

Reply:
One typical pattern is to write a helper function that leverages normal array indexing:
```
def index_along(operand, index, axis=0):
  indices = [slice(None)] * operand.ndim
  indices[axis] = index
  return operand[tuple(indices)]
```

It would be kind of nice to put something like this into JAX somewhere, though I'm not entirely sure where. For now, a little helper function here seems fine.

Maybe I'll try to push it into `np.take` upstream first?

Reply:
I realize now I'm reviving a discussion in a different thread.

@shoyer do you agree with my instinct to resist changing `lax.slice_in_dim`? Or is that dumb because it's a helper function anyway and the new extended behavior is pretty reasonable?

Reply:
I'd be more confident in changing `lax.slice_in_dim` if we could also run our google-internal tests.

But I just did a code search and actually nothing currently calls `lax.slice_in_dim` except our code in lax_numpy.py.

@clemisch I changed my mind about this thread. Let's just go with the PR as-is!